firebase-authexchange/src/main/kotlin/com/google/firebase/authexchange/network/Routes.kt

```diff
@@ -0,0 +1,103 @@
+package com.google.firebase.authexchange.network
+
+import kotlinx.serialization.Serializable
+
+/**
+ * Defines what an HTTP request may look like when interacting with the backend API for
+ * [FirebaseAuthExchange][com.google.firebase.authexchange.FirebaseAuthExchange].
+ */
+@Serializable internal sealed interface Request
+
+/** Defines what an HTTP response may look like when submitting a [Request]. */
+@Serializable internal sealed interface Response
+
+/**
+ * OIDC Credential.
+ *
+ * Useful for apps with external login already set-up, otherwise known as a headless OIDC flow.
+ *
+ * While this isn't the only OIDC flow that the API supports, it's the only flow that the mobile SDK
+ * supports- so there is no base class for credentials.
+ *
+ * @property idToken JWT encoded OIDC token returned from a third party provider
+ */
+@Serializable internal data class ImplicitCredentialsP(val idToken: String)
+
+/**
+ * Auth Token that can be used to access certain Firebase Services.
+ *
+ * This is merely the proto type for the network layer, and is completely separate from the user
+ * facing [AuthExchangeToken][com.google.firebase.authexchange.AuthExchangeToken].
+ *
```

Comment:
what do you mean by a proto type in this case?

Reply:
As in it's a data class used for communicating via protobuf APIs (the backend for auth echange). Although I do think the phrasing gives off the wrong impression. Thoughts on this instead:
```
 * Not to be confused with [AuthExchangeToken][com.google.firebase.authexchange.AuthExchangeToken],
 * this is intended only for network requests.
 ```
 Or maybe just removing it entirely?